pkg/disk/utils.go

```diff
@@ -1165,7 +1215,46 @@ func getAvailableDiskCountFromOpenAPI(ecsClient cloud.ECSInterface, m metadata.M
 	return 0, fmt.Errorf("unexpected DescribeInstanceTypes response for %s: %s", instanceType, response.GetHttpContentString())
 }
 
-func getVolumeCountFromOpenAPI(getNode func() (*v1.Node, error), c cloud.ECSInterface, m metadata.MetadataProvider, dev utilsio.DiskLister) (int, error) {
+func getLingjunAvailableDiskCount(efloClient cloud.EFLOInterface, lingjunID string) (int, error) {
+
+	nodeType, err := getNodeTypeFromEFLOpenAPI(efloClient, lingjunID)
+	if err != nil {
+		return 0, err
+	}
+	klog.InfoS("getLingjunAvailableDiskCount: ", "nodeType", nodeType)
+	if nodeType == "" {
+		return 0, fmt.Errorf("cannot get node type for lingjun %s", lingjunID)
+	}
+	return getAvailableCountFromEFLOpenAPI(efloClient, nodeType)
+}
+
+func getAvailableCountFromEFLOpenAPI(efloClient cloud.EFLOInterface, nodeType string) (int, error) {
+	req := &eflo.DescribeNodeTypeRequest{
+		NodeType: &nodeType,
+	}
+	resp, err := efloClient.DescribeNodeType(req)
+	if err != nil {
+		return 0, fmt.Errorf("DescribeNodeType failed: %w", err)
+	}
+	klog.InfoS("getAvailableCountFromEFLOpenAPI: ", "resp", resp)
+	return int(*resp.Body.DiskQuantity), nil
+}
+
+func getNodeTypeFromEFLOpenAPI(efloClient cloud.EFLOInterface, lingjunID string) (string, error) {
```

Comment:
Can we get this from metadata server? If so (either now or in the future), I hope this can go into cloud/metadata package.